src/common/socket.cpp

```diff
@@ -827,7 +828,7 @@ int32 makeListenBind_tcp(uint32 ip, uint16 port,RecvFunc connect_client)
 	//set_nonblocking(fd, 1);
 
 	server_address.sin_family      = AF_INET;
-	server_address.sin_addr.s_addr = htonl(ip);
+        inet_pton(AF_INET, ip, &(server_address.sin_addr));
```

Comment:
server_address.s_addr = inet_addr(ip);

this is what the map server does, and should be what you do unless it's not possible